internal/dev_server/api/server.go

```diff
@@ -166,7 +166,10 @@ func (s Server) DeleteDevProjectsProjectKeyOverridesFlagKey(ctx context.Context,
 func (s Server) PutDevProjectsProjectKeyOverridesFlagKey(ctx context.Context, request PutDevProjectsProjectKeyOverridesFlagKeyRequestObject) (PutDevProjectsProjectKeyOverridesFlagKeyResponseObject, error) {
 	override, err := model.UpsertOverride(ctx, request.ProjectKey, request.FlagKey, request.Body.String())
 	if err != nil {
-		return nil, err
+		return PutDevProjectsProjectKeyOverridesFlagKey400JSONResponse{InvalidRequestResponseJSONResponse{
+			Code:    "invalid_request",
+			Message: err.Error(),
+		}}, err
```

Comment:
This also took me a long time to figure out but you have to return nil instead of err here.
```suggestion
		return PutDevProjectsProjectKeyOverridesFlagKey400JSONResponse{InvalidRequestResponseJSONResponse{
			Code:    "invalid_request",
			Message: err.Error(),
		}}, nil
```

Reply:
Oh, that's not quite right though. We'll always reply with a 400. I think we need to expose an `ErrFlagNotFound` in the `model` and then we'll need to do like `errors.Is(ErrFlagNotFound)` to see if we 400.

Reply:
fixed - added a new domain error type and when those are returned we'll 400